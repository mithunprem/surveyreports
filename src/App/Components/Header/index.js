import React, { Fragment } from 'react';
import { Navbar, NavbarBrand } from 'reactstrap';

const Header = ({ children }) => {
  return (
    <Fragment>
      <Navbar color="dark" dark expand="xs">
        <NavbarBrand href="/">
          <span>Survey Reports</span>
        </NavbarBrand>
      </Navbar>
      { children }
    </Fragment>
  );
}

export default Header;
