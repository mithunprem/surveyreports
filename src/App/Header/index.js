import React, { Fragment } from 'react';
import { Navbar, NavbarBrand } from 'reactstrap';

function Header () {
  return (
    <Fragment>
      <Navbar color="dark" dark expand="xs">
        <NavbarBrand href="/">
          <span>Survey Reports</span>
        </NavbarBrand>
      </Navbar>
    </Fragment>
  );
}

export default Header;
