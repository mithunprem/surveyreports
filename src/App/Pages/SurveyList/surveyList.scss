.survey-list {
  float: left;
  margin-left: 0;
  margin-right: 0;
}
