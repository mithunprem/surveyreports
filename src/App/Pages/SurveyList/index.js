import React, { Component, Fragment } from 'react';
import { Row, Alert } from 'reactstrap';
import LoadingSpinner from '../../Components/LoadingSpinner';
import Survey from '../../Components/Survey';
import { API_URL } from '../../Constants';
import './surveyList.scss';

export default class SurveyList extends Component {

  state = {
    isSurveyResultsLoading: false,
    surveyResults: []
  };

  componentDidMount() {
    this.setState(
      {
        isSurveyResultsLoading: true
      },
      async () => {
        const surveyResults = await this.fetchSurveyResults();
        this.setState({
          isSurveyResultsLoading: false,
          surveyResults
        });
      }
    );
  }

  fetchSurveyResults() {
    return fetch(`${API_URL}/surveys`)
      .then(response => response.json())
      .then(response => response.survey_results)
      .catch(error => console.error('Error:', error));
  }

  render() {
    const { surveyResults, isSurveyResultsLoading } = this.state;

    if (!surveyResults.length > 0 ) {
      return (
        isSurveyResultsLoading ?
          <LoadingSpinner spinnerText='Loading survey resultsâ€¦' /> :
          <Alert color="danger">
            Survey results failed to load. Please try again !.
          </Alert>
        )
    }

    return (
      <Fragment>
        <Row className="mt-3 mb-3 survey-list">
          {
            surveyResults.map(( survey, index ) => {
              return <Survey key={index} survey={survey} />;
            })
          }
        </Row>
      </Fragment>
    );
  }
}
