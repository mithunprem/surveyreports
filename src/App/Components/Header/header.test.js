import React from 'react';
import { create } from 'react-test-renderer';
import Header from '../../Components/Header';
import SurveyList from '../../Pages/SurveyList';

describe("Header", () => {
  test("should render without throwing an error", () => {
    const header = create(<Header />).toJSON();
    expect(header).toMatchSnapshot();
  });

  test("should render with a child component without throwing an error", () => {
    const header = create(<Header><SurveyList /></Header>).toJSON();
    expect(header).toMatchSnapshot();
  });
});
