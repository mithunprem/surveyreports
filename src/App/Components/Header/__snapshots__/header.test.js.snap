// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render with a child component without throwing an error 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      <span>
        Survey Reports
      </span>
    </a>
  </nav>,
  <div
    className="m-4"
  >
    Loading survey resultsâ€¦
     
    <div
      className="spinner-grow text-dark"
      role="status"
    >
      <span
        className="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>,
]
`;

exports[`Header should render without throwing an error 1`] = `
<nav
  className="navbar navbar-expand navbar-dark bg-dark"
>
  <a
    className="navbar-brand"
    href="/"
  >
    <span>
      Survey Reports
    </span>
  </a>
</nav>
`;
